utils/src/main/kotlin/OrtAuthenticator.kt

```diff
@@ -127,20 +127,29 @@ fun getNetrcAuthentication(contents: String, machine: String): PasswordAuthentic
 
     val iterator = lines.joinToString(" ").split(Regex("\\s+")).iterator()
 
-    var machineFound = false
     var login: String? = null
     var password: String? = null
+    var lastEntry: String? = null
+    val credentialsPerMachine = mutableMapOf<String, Pair<String, String>>()
 
     while (iterator.hasNext()) {
         when (iterator.next()) {
-            "machine" -> machineFound = iterator.hasNext() && iterator.next() == machine
-            "login" -> login = if (machineFound && iterator.hasNext()) iterator.next() else null
-            "password" -> password = if (machineFound && iterator.hasNext()) iterator.next() else null
-            "default" -> machineFound = true
+            "machine" -> lastEntry = iterator.next()
+            "login" -> login = iterator.next()
+            "password" -> password = iterator.next()
+            "default" -> lastEntry = "default"
         }
 
-        if (login != null && password != null) return PasswordAuthentication(login, password.toCharArray())
+        if (login != null && password != null && lastEntry != null) {
```

Comment:
Nit: Maybe check `lastEntry != null` first to make the order nicely match the order in the line below for readability.